packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
   async owner(split: Address): Promise<Address> {
     return this._getSplitV2Contract(split).read.owner()
   }
+
+  async getSplitMetadata(splitAddress: Address): Promise<{ split: SplitV2 }> {
```

Comment:
let's just test this with a couple different providers (not sure who the biggest are, alchemy/infura/quicknode?) curious what performance looks like / if it breaks for any. i know a while ago i was having trouble fetching more than 10k logs in a single request. maybe providers handle that better now, or maybe viem handles it under the hood for us, or maybe we'll need to custom handle that ourselves.

Reply:
yeah, I am not sure what the performance looks like rn, will test after I do some deployments